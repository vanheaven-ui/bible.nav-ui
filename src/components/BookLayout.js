import { Container } from 'react-bootstrap';
import PropTypes from 'prop-types';

const BookLayout = ({
  params, loadVerse, getVerse,
}) => {
  const chapters = params.chapters.filter(chapter => chapter.number !== 'intro');

  return (
    <div className="book-details_wrap">
      <header>
        <h2>
          The book of
          {' '}
          {params.bookName}
        </h2>
        <p>
          Enjoy the
          {' '}
          {chapters.length}
          {' '}
          chapters of this book. Click on each chapter number to get the verses.
        </p>
      </header>
      <Container className="chapter-list">
        <h4>Chapters</h4>
        {' '}
        { params.chapters && chapters.map(chapter => (
          <button
            type="button"
            id={chapter.id}
            key={chapter.id}
            onClick={e => loadVerse(e)}
            className=""
          >
            {chapter.number}
          </button>
        ))}
      </Container>
      <Container className="verse-list">
        <h5>Verses</h5>
        {' '}
        { params.verses && params.verses.map(verse => (
          <button
            type="button"
            id={verse}
            key={verse}
            onClick={e => getVerse(e)}
          >
            {verse}
          </button>
        ))}
      </Container>
    </div>
  );
};

BookLayout.propTypes = {
  params: PropTypes.instanceOf(Object).isRequired,
  loadVerse: PropTypes.func.isRequired,
  getVerse: PropTypes.func.isRequired,
};

export default BookLayout;
