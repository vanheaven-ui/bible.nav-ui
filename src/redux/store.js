import { applyMiddleware, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunkMiddleware from 'redux-thunk';
import rootReducer from './reducers';

const middlewareEnhancer = applyMiddleware(thunkMiddleware);
const composedEnhancer = composeWithDevTools(middlewareEnhancer);

const store = createStore(rootReducer, undefined, composedEnhancer);

export default store;
