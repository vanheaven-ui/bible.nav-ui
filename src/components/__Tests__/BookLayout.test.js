import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import { render, screen, cleanup } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import rootDiv from '../../testHelper';
import store from '../../redux/store';
import '@testing-library/jest-dom';
import BookLayout from '../BookLayout';

describe('BookLayout Component', () => {
  const id = 'JHN';
  const loadVerse = () => '';
  const getVerse = () => '';
  const params = {
    chapters: [
      { bibleId: "55212e3cf5d04d49-01", bookId: "MRK", id: "MRK.1", number: "1", reference: "Mark 1" },
      { bibleId: "55212e3cf5d04d49-01", bookId: "MRK", id: "MRK.2", number: "2", reference: "Mark 2" },
      { bibleId: "55212e3cf5d04d49-01", bookId: "MRK", id: "MRK.3", number: "3", reference: "Mark 3" },
      { bibleId: "55212e3cf5d04d49-01", bookId: "MRK", id: "MRK.4", number: "4", reference: "Mark 4" },
    ],
    bookName: 'John',
    verses: ['1', '2', '3', '4'],
  };

  afterEach(cleanup);

  it('renders without crashing', () => {
    ReactDOM.render(
      <BookLayout params={params} loadVerse={loadVerse} getVerse={getVerse} id={id} />, rootDiv(),
    );
  });

  it('does not change unexpectedly', () => {
    const tree = renderer.create(
      <BookLayout params={params} loadVerse={loadVerse} getVerse={getVerse} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the correct wrapper element', () => {
    const { container } = render(
      <BookLayout params={params} loadVerse={loadVerse} getVerse={getVerse} />,
    );
    expect(container.firstChild.classList.contains('book-details_wrap')).toBe(true);
  });
});
