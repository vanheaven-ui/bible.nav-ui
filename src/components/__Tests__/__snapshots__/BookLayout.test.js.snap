// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookLayout Component does not change unexpectedly 1`] = `
<div
  className="book-details_wrap"
>
  <header>
    <h2>
      The book of
       
      John
    </h2>
    <p>
      Enjoy the
       
      4
       
      chapters of this book. Click on each chapter number to get the verses.
    </p>
  </header>
  <div
    className="chapter-list container"
  >
    <h4>
      Chapters
    </h4>
     
    <button
      className=""
      id="MRK.1"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className=""
      id="MRK.2"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className=""
      id="MRK.3"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className=""
      id="MRK.4"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
  </div>
  <div
    className="verse-list container"
  >
    <h5>
      Verses
    </h5>
     
    <button
      id="1"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      id="2"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      id="3"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      id="4"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
  </div>
</div>
`;
